integration-tests/src/test/java/oracle/kubernetes/operator/ItModelInImageOverride.java

```diff
@@ -0,0 +1,246 @@
+    final String cmName = domainUid + configMapSuffix;
+    final String label = "weblogic.domainUID=" + domainUid;
+
+    TestUtils.createConfigMap(cmName, destDir, domainNS, label);
+  }
+
+  private void modifyDomainYamlWithRestartVersion()
+      throws Exception {
+    String originalYaml =
+        getUserProjectsDir()
+            + "/weblogic-domains/"
+            + domain.getDomainUid()
+            + "/domain.yaml";
+
+    // Modify the original domain yaml to include restartVersion in admin server node
+    DomainCrd crd = new DomainCrd(originalYaml);
+    Map<String, String> objectNode = new HashMap();
+    objectNode.put("restartVersion", "v1.1");
+    crd.addObjectNodeToDomain(objectNode);
+    String modYaml = crd.getYamlTree();
+    LoggerHelper.getLocal().log(Level.INFO, modYaml);
+
+    // Write the modified yaml to a new file
+    Path path = Paths.get(getUserProjectsDir()
+        + "/weblogic-domains/"
+        + domain.getDomainUid(), "modified.domain.yaml");
+    LoggerHelper.getLocal().log(Level.INFO, "Path of the modified domain.yaml :{0}", path.toString());
+    Charset charset = StandardCharsets.UTF_8;
+    Files.write(path, modYaml.getBytes(charset));
+
+    // Apply the new yaml to update the domain crd
+    LoggerHelper.getLocal().log(Level.INFO, "kubectl apply -f {0}", path.toString());
+    ExecResult exec = TestUtils.exec("kubectl apply -f " + path.toString());
+    LoggerHelper.getLocal().log(Level.INFO, exec.stdout());
+    LoggerHelper.getLocal().log(Level.INFO, "Verifying if the domain is restarted");
+    domain.verifyDomainRestarted();
+  }
+
+  private void verifyJdbcOverride() throws Exception {
+    // get domain name
+    StringBuffer cmdKubectlSh = new StringBuffer("kubectl get domain -n ");
+    cmdKubectlSh
+        .append(domainNS)
+        .append(" -o=jsonpath='{.items[0].metadata.name}'");
+
+    LoggerHelper.getLocal().log(Level.INFO, "Command to exec: " + cmdKubectlSh);
+    ExecResult result = TestUtils.exec(cmdKubectlSh.toString());
+    String domainName = result.stdout();
+    LoggerHelper.getLocal().log(Level.INFO, "Domain name is: " + domainName);
+
+    // check JDBC DS override
+    cmdKubectlSh = new StringBuffer("kubectl -n ");
+    cmdKubectlSh
+        .append(domainNS)
+        .append(" exec -it ")
+        .append(domain.getDomainUid())
+        .append("-")
+        .append(domain.getAdminServerName())
+        .append(" -- bash -c 'cd /u01/oracle/user_projects/domains/")
+        .append(domainName)
+        .append("/config/jdbc/")
+        .append(" && grep -R jdbc/generic1")
+        .append("'");
+
+    LoggerHelper.getLocal().log(Level.INFO, "Command to exec: " + cmdKubectlSh);
+    result = TestUtils.exec(cmdKubectlSh.toString());
+    LoggerHelper.getLocal().log(Level.INFO, "JDBC DS info from server pod: " + result.stdout());
+
+    Assumptions.assumeTrue(result.stdout().contains("<jndi-name>jdbc/generic1</jndi-name>"),
```

Comment:
should this be an assumption or an assertion?

Reply:
You are right, should be assertion so that the failures can be registered as a failed test. fixed